### PR TITLE
fix(monitoring): better rendering of notification config

### PR DESCRIPTION
**Pull request by cgagnaire**
+ fix asort function : SORT_NATURAL constant implemented since PHP 5.4, replaced by SORT_STRING for PHP 5.3 compatibility,
+ fix infinite loop : "continue" instead of "break",
+ handling of disabled notification on hosts and services (with templates inheritance),
+ handling of disabled contacts and contact groups,
+ handling of disabled notification on contacts,
+ fix case where service has no template and was ignored.

### CODE REVIEW

File: www/include/monitoring/common-Func.php [hunk 1]
```diff
@@ -89,41 +89,66 @@ function get_notified_infos_for_host($hostId)
     $loop = array();    
     $stack = array($hostId);
     $hosts = array();
-    $results = array('contacts' => array(), 'contactGroups' => array());
-    $stopReading = array('contacts' => 0, 'contactGroups' => 0);
+    $results = array('contacts' => array(), 'contactGroups' => array(), 'enableNotif' => 2);
+    $stopReading = array('contacts' => 0, 'contactGroups' => 0, 'notifParam' => 0, 'contactAdditiveInheritance' => 0, 'cgAdditiveInheritance' => 0);
```
Comment (kduret) [inline www/include/monitoring/common-Func.php]: This line exceeds 120 characters

File: www/include/monitoring/common-Func.php [hunk 2]
```diff
@@ -179,66 +211,98 @@ function get_notified_infos_for_service($serviceId, $hostId)
 {
     global $pearDB;
 
-    $loop = array();    
-    $results = array('contacts' => array(), 'contactGroups' => array());
-    $stopReading = array('contacts' => 0, 'contactGroups' => 0);
+    $loop = array();
+    $results = array('contacts' => array(), 'contactGroups' => array(), 'enableNotif' => 2);
+    $stopReading = array('contacts' => 0, 'contactGroups' => 0, 'notifParam' => 0, 'contactAdditiveInheritance' => 0, 'cgAdditiveInheritance' => 0);
```
Comment (kduret) [inline www/include/monitoring/common-Func.php]: This line exceeds 120 characters

File: www/include/monitoring/common-Func.php [hunk 3]
```diff
@@ -179,66 +211,98 @@ function get_notified_infos_for_service($serviceId, $hostId)
...
         
         /*
          * Manage contact inheritance
          */
-        $contactGroups = getContactgroupsForService($serviceId);
-        $contacts = getContactsForService($serviceId);
-        
-        if ($stopReading['contacts'] == 0) {
-            $results['contacts'] = $results['contacts'] + $contacts;
+        if (!is_null($notifParam['service_use_only_contacts_from_host']) && $notifParam['service_use_only_contacts_from_host'] == 1) {
```
Comment (kduret) [inline www/include/monitoring/common-Func.php]: This line exceeds 120 characters

**Comment by cgagnaire** [comment]
Add https://github.com/centreon/centreon/pull/5976 modification from @jben94

**Comment by lpinsivy** [comment]
So we can close PR #5976 @cgagnaire ?

**Pull request closed without merging**